### DISCUSSION TITLE
Determine which provider was modified during build

### DISCUSSION THREAD

**Discussion by dickermoshe** [rrousselGit/riverpod#3822] [General]
We all have seen this error:
<img width="623" alt="image" src="https://github.com/user-attachments/assets/104b9710-9b35-42ae-82ef-a84721d5cfee">

But I've got lots of widgets and tons of providers, is there a way to get a better error message than this?

**Comment by rrousselGit** [comment]
The error is from Flutter.

The StackTrace should point you to the problematic code

**Reply by dickermoshe** [reply]
Good point, I see that in the debugger:
<img width="810" alt="image" src="https://github.com/user-attachments/assets/6d0c3569-de86-4219-98bb-1c8b2bf27da8">

But it shouldn't it also be shown in the error message?
When running an app in debug mode, you only see the message, the stacktrace is not shown in the terminal until you press "Continue"

<img width="396" alt="image" src="https://github.com/user-attachments/assets/35f7b370-6454-4ea2-b97b-e4fe2e86e587">

It would nice if the message could extract this from the `err` object.

@rrousselGit Thank you so much for this package, it's a gem!

**Reply by rrousselGit** [reply]
The stacktrace should be visible in vscode when paused at an error. It's in the debugging panel.

As for the error's toString, I'm not responsible for this error, so I have no control over it. I'm not sure why they would or wouldn't show the ErrorDescription

**Reply by dickermoshe** [reply]
Incorrect, the error being shown is from riverpod.
It would be nice if this err was put directly in the message:
<img width="716" alt="image" src="https://github.com/user-attachments/assets/ff42e21b-196c-4ee4-83d1-87c6c73f10fa">

**Reply by dickermoshe** [reply]
https://github.com/rrousselGit/riverpod/blob/9e62837a9fb6741dc40728c6e28d0fd9d62452e3/packages/flutter_riverpod/lib/src/framework.dart#L348-L377

**Reply by dickermoshe** [reply]
I guess the flutter error message ain't great to begin with.
